### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by calebmshafer**
The iModel.js org has access to the Code Scanning beta offered by GitHub.  Setup a basic scan that will run on every PR and on a weekly basis.

This is very similar, if not the same, as what will be found on LGTM, https://lgtm.com/projects/g/imodeljs/viewer-components-react/?mode=list. 

Learn more about how to config it and monitor the results [here](https://docs.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code).

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
+      if: ${{ github.event_name == 'pull_request' }}
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      # Override language selection by uncommenting this and choosing your languages
+      # with:
+      #   languages: go, javascript, csharp, python, cpp, java
```
Comment (aruniverse) [inline .github/workflows/codeql.yml:35]: this monorepo will only have js/ts files so maybe we should explicitly select which languages to use.

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
```
Comment (aruniverse) [inline .github/workflows/codeql.yml:39]: looks like we can remove the autobuild step

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (calebmshafer) [reply .github/workflows/codeql.yml:35]: Yea, I went back and forth on cleaning up some of these defaults. I wanted to see how this went before going through and cleaning it up but you're right.

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (calebmshafer) [reply .github/workflows/codeql.yml:39]: good point, I'll remove it

**Pull request merged**